content/blog/super-charging-our-github-action-with-the-pulumi-automation-api/index.md

```diff
@@ -0,0 +1,197 @@
+        with:
+          node-version: ${{ matrix.node_version }}
+      - name: Install pulumi
+        uses: pulumi/action-install-pulumi-cli@v1.0.1
+      - run: npm install
+      - uses: actions/pulumi@v2
+        env:
+          PULUMI_CONFIG_PASSPHRASE: not-a-secret
+        with:
+          command: preview
+          cloud-url: file://~
+          stack-name: dev
+```
+
+### Cloud Interaction
+
+One of the things that the previous container did on your behalf was that if you had a `GOOGLE_CREDENTIALS` environment
+variable set then it would run the following commands on your behalf:
+
+```shell
+gcloud auth activate-service-account --key-file=$GOOGLE_APPLICATION_CREDENTIALS
+gcloud --quiet auth configure-docker $GOOGLE_DOCKER_HOSTNAME_LIST
+```
+
+With this new action, we wanted to defer the authentication of your cloud to be in your hands. So if you want to authenticate
+with AWS / Azure / Google Cloud then we suggest you use their official actions to do so:
+
+{{% chooser cloud "aws,azure,gcp" / %}}
+
+{{% choosable cloud aws %}}
+
+```yaml
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-region: ${{ secrets.AWS_REGION }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+```
+
+{{< /choosable >}}
+
+{{% choosable cloud azure %}}
+
+```yaml
+    - uses: azure/login@v1
+      with:
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+```
+
+{{< /choosable >}}
+
+{{% choosable cloud gcp %}}
+
+```yaml
+      - uses: google-github-actions/setup-gcloud@master
+        with:
+          service_account_key: ${{ secrets.GCP_KEY }}
+          project_id: ${{ env.PROJECT_ID }}
+          export_default_credentials: true
+```
+
+{{< /choosable >}}
+
+We suggest that any sensitive environment variables be referenced using
+[GitHub Secrets](https://developer.github.com/actions/creating-workflows/storing-secrets/),
+and consume them using
+[the `secrets` attribute](https://developer.github.com/actions/creating-workflows/workflow-configuration-options/#actions-attributes)
+on your workflow's action.
+
+### Faster
+
+The GitHub Action is much smaller and no longer needs to download a (very!) large Docker container. In the dockerfile based
+action, up to 3 minutes was spent downloading the container. The new action is much faster as it no longer has this
+overhead. Let's look at the comparison:
+
+#### Docker based Action (v1)
+
+![docker based action](old-action.png)
+
+#### Automation API Supercharged Action
+
+![Automation API super charged based action](new-action.png)
+
+We are eager to hear your feedback on how we can improve your experience with this feature. If you haven’t already signed up for our Community Slack, it’s quick and easy! You can join in on conversations you like and get help from other community members, as well as the Pulumi Team. That’s it for now! 👋
```

Comment:
Is there a very simple page we can link to for folks who want to use it? For instance, is https://www.pulumi.com/docs/guides/continuous-delivery/github-actions/ the right place? I would assume 10X more folks just want to kick the tires and try it, compared to those who want to know how it works under the hood.

Reply:
100% agree. Would love to end with a CTA like:

```suggestion
The new Pulumi GitHub Action is ready for you to use. Here's how to get started:

 - Already use the v1 Pulumi GitHub Action? Migrate to the new one with our [migration guide](TODO-link).
 - New to Pulumi and GitHub Actions? Use our [getting started guide](TODO-link) to set up your first Pulumi workflow.
 
We're eager to hear your feedback on this new Action. If you haven’t already signed up for our Community Slack, it’s quick and easy! You can join in on conversations you like and get help from other community members, as well as the Pulumi Team. That’s it for now! 👋
```

Reply:
way better! Thanks :)